Test Building at UGent: Zen3/A100

laraPPr opened the pull request.

laraPPr commented:
bot: help

laraPPr commented:
bot: help

laraPPr commented:
bot: help

laraPPr commented:
bot show_config instance: eessi-bot-vsc-ugent

laraPPr commented:
bot show_config instance:eessi-bot-vsc-ugent

laraPPr commented:
bot show_config

trz42 commented:
> bot show_config instance:eessi-bot-vsc-ugent

missing ':' after 'bot'?

laraPPr commented:
bot: show_config instance:eessi-bot-vsc-ugent

trz42 commented:
> @bedroge the riscv bot seems to be doing double duty maybe check that out. [#842 (comment)](https://github.com/EESSI/software-layer/pull/842#issuecomment-2547953008)

It's suspicious that it only replies once when the PR is opened, but twice when it receives "command" via a comment. Maybe worth to check the logs.

laraPPr commented:
> > @bedroge the riscv bot seems to be doing double duty maybe check that out. [#842 (comment)](https://github.com/EESSI/software-layer/pull/842#issuecomment-2547953008)
> 
> It's suspicious that it only replies once when the PR is opened, but twice when it receives "command" via a comment. Maybe worth to check the logs.

Yes I was debugging. I had a typo in the smee link and than their was an issue with the target_map. Those two are now resolved which is why the ugent bot only started reacting later.

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/15437029`
> 
> date	job status	comment
> Dec 17 09:55:32 UTC 2024	submitted	job id `15437029` awaits release by job manager

I think the job manager will fail parsing squeue but let's see

laraPPr commented:
Job was submitted 
```
[EESSIUSER@bot machine]$ squeue

CLUSTER: accelgor

             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)

          15437029  accelgor     prod EESSIUSER PD       0:00      1 (JobHeldUser)
```

laraPPr commented:
> > New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/15437029`
> > date	job status	comment
> > Dec 17 09:55:32 UTC 2024	submitted	job id `15437029` awaits release by job manager
> 
> I think the job manager will fail parsing squeue but let's see

It did 
```
export GITHUB_APP_SECRET_TOKEN=$(cat /scratch/gent/485/vsc48506/ugent-config-202411/webhook_secret)

export GITHUB_TOKEN=$(cat /scratch/gent/485/vsc48506/ugent-config-202411/access_token-20241209)

./job_manager.sh

Configuration check: PASSED

job manager just started, logging to '/scratch/gent/485/vsc48506/eessi_bot_job_manager.log', processing job ids ''

Traceback (most recent call last):

  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main

    return _run_code(code, main_globals, None,

  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code

    exec(code, run_globals)

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 733, in <module>

    main()

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 692, in main

    is_bot_job = job_manager.process_new_job(current_jobs[nj])

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 280, in process_new_job

    scontrol_output, scontrol_err, scontrol_exitcode = run_cmd(

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/tools/__init__.py", line 59, in run_cmd

    raise RuntimeError(error_msg)

RuntimeError: run_cmd(): Error running '/usr/bin/scontrol --oneliner show jobid 15437029' in 'None

           stdout ''

           stderr 'slurm_load_jobs error: Invalid job id specified

'

           exit code 1
```

laraPPr commented:
> > New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/15437029`
> > date	job status	comment
> > Dec 17 09:55:32 UTC 2024	submitted	job id `15437029` awaits release by job manager
> 
> I think the job manager will fail parsing squeue but let's see

it did 
```
job manager just started, logging to '/scratch/gent/485/vsc48506/eessi_bot_job_manager.log', processing job ids ''

Traceback (most recent call last):

  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main

    return _run_code(code, main_globals, None,

  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code

    exec(code, run_globals)

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 733, in <module>

    main()

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 692, in main

    is_bot_job = job_manager.process_new_job(current_jobs[nj])

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/eessi_bot_job_manager.py", line 280, in process_new_job

    scontrol_output, scontrol_err, scontrol_exitcode = run_cmd(

  File "/kyukon/scratch/gent/485/vsc48506/eessi-bot-software-layer/tools/__init__.py", line 59, in run_cmd

    raise RuntimeError(error_msg)

RuntimeError: run_cmd(): Error running '/usr/bin/scontrol --oneliner show jobid 15437029' in 'None

           stdout ''

           stderr 'slurm_load_jobs error: Invalid job id specified

'

           exit code 1
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software
# this has now a walltime of 30min to see if it will start sooner and I can see if the squeue parsing by the job_manager works

laraPPr commented:
This does not look correctly
```
[architecture]

software_subdir = x86_64/amd/zen3

os_type = linux

accelerator =
```

laraPPr commented:
> This does not look correctly
> 
> ```
> [architecture]
> 
> software_subdir = x86_64/amd/zen3
> 
> os_type = linux
> 
> accelerator =
> ```

Ah this is probably because I did not set `accel:nvidia/cc80` in the command? @trz42

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
@casparvl  the tests are failing because the following tag is set ` %scale=1_node` I'm only asking for a 1_4_node so all the test fail.

laraPPr commented:
Ah the job is now running but the bot does not seem to notice

laraPPr commented:
output of reframe:
```
Running tests: reframe --tag CI --tag 1_node --nocolor -n EESSI_OSU_Micro_Benchmarks -n EESSI_LAMMPS --run

[ReFrame Setup]

  version:           4.6.2

  command:           '/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/ReFrame/4.6.2/bin/reframe --tag CI --tag 1_node --nocolor -n EESSI_OSU_Micro_Benchmarks -n EESSI_LAMMPS --run'

  launched by:       vsc48506@node3903.accelgor.os

  working directory: '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software'

  settings files:    '<builtin>', '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_config_bot.py'

  check search path: (R) '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/EESSI-test-suite/eessi/testsuite/tests'

  stage directory:   '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage'

  output directory:  '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/output'

  log files:         '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/logs/reframe_20241217_144635.log'



^[[33mWARNING: skipping test file '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/EESSI-test-suite/eessi/testsuite/tests/apps/MetalWalls.py': module not found error: EESSI-test-suite/eessi/testsuite/tests/apps/MetalWalls.py:31: No module named 'hpctestlib.sciapps.metalwalls'

from hpctestlib.sciapps.metalwalls.benchmarks import MetalWallsCheck

 (rerun with '-v' for more information)^[[0m

^[[33mWARNING: skipping test file '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/EESSI-test-suite/eessi/testsuite/tests/apps/gromacs.py': value error: EESSI-test-suite/eessi/testsuite/utils.py:124: EESSI test-suite cannot handle duplicate modules. Please make sure that only one is available on your system. The following modules have a duplicate on your system: ['GROMACS/2024.4-foss-2023b']

        raise ValueError(err_msg)

 (rerun with '-v' for more information)^[[0m

^[[33mWARNING: skipping test file '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/EESSI-test-suite/eessi/testsuite/tests/apps/tensorflow/tensorflow.py': value error: EESSI-test-suite/eessi/testsuite/utils.py:124: EESSI test-suite cannot handle duplicate modules. Please make sure that only one is available on your system. The following modules have a duplicate on your system: ['TensorFlow/2.13.0-foss-2023a']

        raise ValueError(err_msg)

 (rerun with '-v' for more information)^[[0m

"slurm-15437065.out" 1312L, 114180B
[==========] Running 14 check(s)

[==========] Started on Tue Dec 17 14:46:49 2024+0000



[----------] start processing checks

[ RUN      ] EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:accelgor+default

^[[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.^[[0m

[ RUN      ] EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:accelgor+default

^[[33mWARNING: hooks.req_memory_per_node does not support the scheduler you configured (local). The test will run, but since it doesn't request the required amount of memory explicitely, it may result in an out-of-memory error. Please expand the functionality of hooks.req_memory_per_node for your scheduler.^[[0m

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /0d62e513 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /31ac6ab9 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /f3be40a2 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /26bfa764 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /10e66fba @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /5be57ae7 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /07f8b5cd @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /c8c9aff5 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /9795e491 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /74c4cdd7 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /48da21c5 @BotBuildTests:accelgor+default

[ RUN      ] EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /1b8c1ca2 @BotBuildTests:accelgor+default

[     FAIL ] ( 1/14) EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/29Aug2024-foss-2023b-kokkos /aeb2d9df @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_LAMMPS_lj_aeb2d9df'

[     FAIL ] ( 2/14) EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos /04ff9ece @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_LAMMPS_lj_04ff9ece'

[     FAIL ] ( 3/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /0d62e513 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_0d62e513'

[     FAIL ] ( 4/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /31ac6ab9 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_31ac6ab9'

[     FAIL ] ( 5/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /f3be40a2 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_f3be40a2'

[     FAIL ] ( 6/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /26bfa764 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_26bfa764'

[     FAIL ] ( 7/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /10e66fba @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_10e66fba'

[     FAIL ] ( 8/14) EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_alltoall %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /5be57ae7 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_5be57ae7'

[     FAIL ] ( 9/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /07f8b5cd @BotBuildTests:accelgor+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_pt2pt_07f8b5cd'

[       OK ] (10/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /c8c9aff5 @BotBuildTests:accelgor+default

P: latency: 0.11 us (r:0, l:None, u:None)

[       OK ] (11/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_latency %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /9795e491 @BotBuildTests:accelgor+default

P: latency: 0.11 us (r:0, l:None, u:None)

[     FAIL ] (12/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu /74c4cdd7 @BotBuildTests:accelgor+default

==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_pt2pt_74c4cdd7'

[       OK ] (13/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu /48da21c5 @BotBuildTests:accelgor+default

P: bandwidth: 30583.88 MB/s (r:0, l:None, u:None)

[       OK ] (14/14) EESSI_OSU_Micro_Benchmarks_pt2pt %benchmark_info=mpi.pt2pt.osu_bw %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu /1b8c1ca2 @BotBuildTests:accelgor+default

P: bandwidth: 30870.89 MB/s (r:0, l:None, u:None)

[----------] all spawned checks have finished



[  FAILED  ] Ran 14/14 test case(s) from 14 check(s) (10 failure(s), 0 skipped, 0 aborted)

[==========] Finished on Tue Dec 17 14:54:37 2024+0000

================================================================================

SUMMARY OF FAILURES

--------------------------------------------------------------------------------

FAILURE INFO for EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/29Aug2024-foss-2023b-kokkos (run: 1/1)

  * Description:

  * System partition: BotBuildTests:accelgor

  * Environment: default

  * Stage directory: /kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_LAMMPS_lj_aeb2d9df

  * Node list: node3903.accelgor.os

  * Job type: local (id=894)

  * Dependencies (conceptual): []

  * Dependencies (actual): []

  * Maintainers: []

  * Failing phase: sanity

  * Rerun with '-n /aeb2d9df -p default --system BotBuildTests:accelgor -r'

  * Reason: sanity error: not enough matches of pattern '^  using (?P<threads>[0-9]+) OpenMP thread\\(s\\) per MPI task' in file 'rfm_job.out' so as to extract item 0

--- rfm_job.out (first 10 lines) ---

EESSI_CVMFS_REPO: /cvmfs/software.eessi.io

EESSI_SOFTWARE_SUBDIR: x86_64/amd/zen3

FULL_MODULEPATH:

--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

--------------------------------------------------------------------------

There are not enough slots available in the system to satisfy the 48

slots that were requested by the application:



  lmp



Either request fewer slots for your application, or make more slots
available for use.



A "slot" is the Open MPI term for an allocatable unit where we can

--- rfm_job.err ---

--------------------------------------------------------------------------------

FAILURE INFO for EESSI_LAMMPS_lj %scale=1_node %device_type=cpu %module_name=LAMMPS/2Aug2023_update2-foss-2023a-kokkos (run: 1/1)

  * Description:

  * System partition: BotBuildTests:accelgor

  * Environment: default

  * Stage directory: /kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_LAMMPS_lj_04ff9ece

  * Node list: node3903.accelgor.os

  * Job type: local (id=907)

  * Dependencies (conceptual): []

  * Dependencies (actual): []

  * Maintainers: []

  * Failing phase: sanity

  * Rerun with '-n /04ff9ece -p default --system BotBuildTests:accelgor -r'

  * Reason: sanity error: not enough matches of pattern '^  using (?P<threads>[0-9]+) OpenMP thread\\(s\\) per MPI task' in file 'rfm_job.out' so as to extract item 0

--- rfm_job.out (first 10 lines) ---

EESSI_CVMFS_REPO: /cvmfs/software.eessi.io

EESSI_SOFTWARE_SUBDIR: x86_64/amd/zen3

FULL_MODULEPATH:

--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

--------------------------------------------------------------------------

There are not enough slots available in the system to satisfy the 48

slots that were requested by the application:



  lmp



Either request fewer slots for your application, or make more slots

available for use.



A "slot" is the Open MPI term for an allocatable unit where we can

--- rfm_job.err ---

--------------------------------------------------------------------------------

FAILURE INFO for EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.4-gompi-2024a %device_type=cpu (run: 1/1)

  * Description:

  * System partition: BotBuildTests:accelgor

  * Environment: default

  * Stage directory: /kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_0d62e513

  * Node list: node3903.accelgor.os

  * Job type: local (id=921)

  * Dependencies (conceptual): []

  * Dependencies (actual): []

  * Maintainers: []

  * Failing phase: sanity
  * Rerun with '-n /0d62e513 -p default --system BotBuildTests:accelgor -r'

  * Reason: sanity error: pattern '^8.*Pass' not found in 'rfm_job.out'

--- rfm_job.out (first 10 lines) ---

--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

Lmod has detected the following error: The following module(s) are unknown:

"OSU-Micro-Benchmarks/7.4-gompi-2024a"



Please check the spelling or version number. Also try "module spider ..."

It is also possible your cache file is out-of-date; it may help to try:

  $ module --ignore_cache load "OSU-Micro-Benchmarks/7.4-gompi-2024a"



Also make sure that all modulefiles written in TCL start with the string

#%Module



--- rfm_job.err ---

--------------------------------------------------------------------------------

FAILURE INFO for EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.2-gompi-2023b %device_type=cpu (run: 1/1)

  * Description:

  * System partition: BotBuildTests:accelgor

  * Environment: default

  * Stage directory: /kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_31ac6ab9

  * Node list: node3903.accelgor.os

  * Job type: local (id=1982)

  * Dependencies (conceptual): []

  * Dependencies (actual): []

  * Maintainers: []

  * Failing phase: sanity

  * Rerun with '-n /31ac6ab9 -p default --system BotBuildTests:accelgor -r'

  * Reason: sanity error: pattern '^8.*Pass' not found in 'rfm_job.out'

--- rfm_job.out (first 10 lines) ---

--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

--------------------------------------------------------------------------

There are not enough slots available in the system to satisfy the 48

slots that were requested by the application:



  osu_allreduce



Either request fewer slots for your application, or make more slots

available for use.



A "slot" is the Open MPI term for an allocatable unit where we can

--- rfm_job.err ---

--------------------------------------------------------------------------------

FAILURE INFO for EESSI_OSU_Micro_Benchmarks_coll %benchmark_info=mpi.collective.osu_allreduce %scale=1_node %module_name=OSU-Micro-Benchmarks/7.1-1-gompi-2023a %device_type=cpu (run: 1/1)

  * Description:

  * System partition: BotBuildTests:accelgor
  * Environment: default

  * Stage directory: /kyukon/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/event_c9e875c0-bc82-11ef-93fa-526f4679e488/run_000/linux_x86_64_amd_zen3/eessi.io-2023.06-software/reframe_runs/stage/BotBuildTests/accelgor/default/EESSI_OSU_Micro_Benchmarks_coll_f3be40a2

  * Node list: node3903.accelgor.os

  * Job type: local (id=1990)

  * Dependencies (conceptual): []

  * Dependencies (actual): []

  * Maintainers: []

  * Failing phase: sanity

  * Rerun with '-n /f3be40a2 -p default --system BotBuildTests:accelgor -r'

  * Reason: sanity error: pattern '^8.*Pass' not found in 'rfm_job.out'

--- rfm_job.out (first 10 lines) ---

--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

--------------------------------------------------------------------------

There are not enough slots available in the system to satisfy the 48

slots that were requested by the application:



  osu_allreduce



Either request fewer slots for your application, or make more slots

available for use.



A "slot" is the Open MPI term for an allocatable unit where we can

```

casparvl commented:
> @casparvl the tests are failing because the following tag is set ` %scale=1_node` I'm only asking for a 1_4_node so all the test fail.

Hm, I guess I didn't think of that when I made the logic for the AWS/Azure clusters (where we always get exclusive nodes). I'll need to see where exactly we set this, and if it can easily be tuned.

casparvl commented:
Well, [this](https://github.com/EESSI/software-layer/blob/65bd38ff7dc74e1617d302f653d206e48165a98a/test_suite.sh#L220C1-L220C43) is where that `1_node`  comes from. But I'm not sure if changing that to `1_4_node` would be the solution (even if we could do it specifically only for your bot). Maybe it should actually be a change in the ReFrame config file - which by the way doesn't account for the potential presence of a GPU either.

laraPPr commented:
It is also not running the GPU ReFrame tests because this is not set in the configuration file

laraPPr commented:
> ReFrame also finds our local modules like ` OSU-Micro-Benchmarks/7.4-gompi-2024a` and Tensorflow and Gromacs (they are skipped) But they can't get to them in the rfm.job. SO it fails. So I need to look where I best insert and `module purge --force` somewhere best as soon as possible as I think this could also cause trouble with `build.sh`

Maybe that could be done with Sam's new feature?

casparvl commented:
Ok, so... the template reframe config file https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/reframe_config_bot.py.tmpl only has `__RFM_PARTITION__` and `__MEM_PER_NODE__` that are still set by [this logic](https://github.com/EESSI/software-layer/blob/65bd38ff7dc74e1617d302f653d206e48165a98a/test_suite.sh#L177) in the `test_suite.sh` script. For the rest, it relies on ReFrame's autodetection to determine the CPU topology in the current partition. It _seems_ that autodetection then determines you have 48 core, which is _physically_ true. But your CGROUP only allows access to 12.

I checked, and even running `--detect-host-topology` locally in an allocation will return the _physical_ topology, not what you have access to in your cgroup:

```
$ numactl --show
policy: default
preferred node: current
physcpubind: 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31
cpubind: 0 1
nodebind: 0 1
membind: 0 1 2 3 4 5 6 7
preferred:

$ reframe --detect-host-topology | tail
          "0xff0000000000000000000000000000000000000000000000"
        ]
      }
    ]
  },
  "num_cpus": 192,
  "num_cpus_per_core": 1,
  "num_cpus_per_socket": 96,
  "num_sockets": 2
}
```
So, that's not helpful either.

I'm really not sure how best to solve this. Should we take the output from `reframe --detect-host-topology` and try to clean it up based on our current `CGROUP`? That sounds... messy.

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software
`# This is a test to see if build.sh will find the modules installed on the local cluster and not build because it can find that installation`

bedroge commented:
> @bedroge the riscv bot seems to be doing double duty maybe check that out. [#842 (comment)](https://github.com/EESSI/software-layer/pull/842#issuecomment-2547953008)

I restarted the event handler and smee, looks like that solved the issue.

casparvl commented:
> ReFrame also finds our local modules like  OSU-Micro-Benchmarks/7.4-gompi-2024a and Tensorflow and Gromacs (they are skipped) But they can't get to them in the rfm.job. SO it fails. So I need to look where I best insert and module purge --force somewhere best as soon as possible as I think this could also cause trouble with build.sh

This is _weird_ by the way, since with the local spawner, the rfm.job essentially runs locally. So how can the runtime find your local modules, but then the job doesn't, even though that is spawned locally? :|

casparvl commented:
It is probably good to mask all local modules in some way for the _entire_ job (i.e. build _and_ test phase). Not sure if that's easily possible.

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/15437096`
> 
> date	job status	comment
> Dec 17 16:00:46 UTC 2024	submitted	job id `15437096` awaits release by job manager
> Dec 17 16:01:58 UTC 2024	released	job awaits launch by Slurm scheduler
> Dec 17 16:46:15 UTC 2024	running	job `15437096` is running
> Dec 17 17:14:59 UTC 2024	finished	
> 🤷 UNKNOWN _(click triangle for detailed information)_
> Dec 17 17:14:59 UTC 2024	test result	
> 🤷 UNKNOWN _(click triangle for detailed information)_

It started the build so the in the build.sh script it does not seem to find the local builds. Will add a check for modulepath in the two scripts as will also do the quick and dirty solution of the test-suite

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software
`building ptm and added a check to see what is in the $MODULEPATH and made small changes to the test-suite so that the tests will run`

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80
building ptm and added a check to see what is in the $MODULEPATH and made small changes to the test-suite so that the tests will run

laraPPr commented:
The local module path should not be set 
Where I put the check right now it also seems early enough to do the `module purge --force` which unsets all the relevant module paths. @boegel and I were discussing that we want to use the local module of awscli but I don't think that is needed within the job environment?

trz42 commented:
> The local module path should not be set
> Where I put the check right now it also seems early enough to do the `module purge --force` which unsets all the relevant module paths. @boegel and I were discussing that we want to use the local module of awscli but I don't think that is needed within the job environment?
> `bot/build.sh: MODULEPATH='/apps/gent/RHEL9/zen3-ampere-ib/modules/all:/etc/modulefiles/vsc'`

The `aws` command is used by the event handler on the host where it is running (not inside a job, that's correct). The bot expects `aws` is in the `$PATH` and the `aws` is actually called by another script `scripts/eessi-upload-to-staging`. So a quick-and-dirty work around could be to modify that script and load the local module of awscli.

laraPPr commented:
> > The local module path should not be set
> > Where I put the check right now it also seems early enough to do the `module purge --force` which unsets all the relevant module paths. @boegel and I were discussing that we want to use the local module of awscli but I don't think that is needed within the job environment?
> > `bot/build.sh: MODULEPATH='/apps/gent/RHEL9/zen3-ampere-ib/modules/all:/etc/modulefiles/vsc'`
> 
> The `aws` command is used by the event handler on the host where it is running (not inside a job, that's correct). The bot expects `aws` is in the `$PATH` and the `aws` is actually called by another script `scripts/eessi-upload-to-staging`. So a quick-and-dirty work around could be to modify that script and load the local module of awscli.

Ah the `module purge --force` needs to be in the job environment so than it is best to add it to the  build.sh script as soon as possible maybe the place where I now added the check. Which would be just a quick and dirty solution. But I'm also added some other quick and dirty tricks for ReFrame I can do the same for the job environment. And than we can still use the awscli from the local module environment on the host.

laraPPr commented:
bot: help

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` and accelerator `nvidia/cc80` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2024.12/pr_842/15437288`
> 
> date	job status	comment
> Dec 19 10:18:08 UTC 2024	submitted	job id `15437288` awaits release by job manager
> Dec 19 10:18:16 UTC 2024	released	job awaits launch by Slurm scheduler
> Dec 19 10:42:22 UTC 2024	running	job `15437288` is running
> Dec 19 10:58:52 UTC 2024	finished	
> 😁 SUCCESS _(click triangle for details)_
> Dec 19 10:58:52 UTC 2024	test result	
> 😢 FAILURE _(click triangle for details)_

It did not skip the LAMMPS test even though it is double. I think the ReFrame did not see the EESSI GPU builds. I'm guessing that if I unset the module path it will only do the OSU tests which is skipping.

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented on bot/test.sh:
```diff
@@ -123,6 +123,11 @@ mkdir -p ${STORAGE}
 JOB_STORAGE=$(mktemp --directory --tmpdir=${STORAGE} bot_job_tmp_XXX)
 echo "bot/test.sh: created unique base tmp storage directory at ${JOB_STORAGE}"
 
+echo "bot/test.sh: MODULEPATH='${MODULEPATH}'"
```
```suggestion
echo "bot/test.sh: MODULEPATH='${MODULEPATH}'"
echo "bot/test.sh: does it know that it is on a UGent cluster? HPCUGENT_FAMILY_CLUSTER='${HPCUGENT_FAMILY_CLUSTER}'"
```

laraPPr commented:
Is only seeing the modules in the CPU path which is only OSU. and it skipped those tests because their is only 1 GPU in the allocation.

laraPPr commented:
I think this is setting the PATH only to the CPU PATH https://github.com/laraPPr/software-layer/blob/d3227b6657991cf6dda4c12650d5ab099cb3dd96/test_suite.sh#L76C10-L80

laraPPr commented:
The sourcing of EESSI in test_suite.sh fails to detect the gpus 
```
Running 'export EESSI_COMPAT_LAYER_DIR=/cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64; export EESSI_VERSION_OVERRIDE=2023.06; export EESSI_CVMFS_REPO_OVERRIDE=/cvmfs/software.eessi.io; export EESSI_SOFTWARE_SUBDIR_OVERRIDE=x86_64/amd/zen3; export SLURM_JOB_ID=15437324; git clone https://github.com/EESSI/test-suite EESSI-test-suite' in EESSI (/cvmfs/software.eessi.io) 2023.06 compatibility layer environment...

Entering Gentoo Prefix /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64

Cloning into 'EESSI-test-suite'...

Leaving Gentoo Prefix with exit status 0

>> Setting up environment...

EESSI_SOFTWARE_SUBDIR_OVERRIDE: x86_64/amd/zen3

Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!

archdetect says x86_64/amd/zen3

archdetect could not detect any accelerators

Using x86_64/amd/zen3 as software subdirectory.
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
```
--- rfm_job.out ---

--- rfm_job.err (first 10 lines) ---

Lmod has detected the following error:

You requested to load UCX-CUDA  which relies on the CUDA runtime environment

and driver libraries. In order to be able to use the module, you will need to

make sure EESSI can find the GPU driver libraries on your host system. You can

override this check by setting the environment variable

EESSI_OVERRIDE_GPU_CHECK but the loaded application will not be able to execute

on your system.

For more information on how to do this, see https://www.eessi.io/docs/gpu/.
```

laraPPr commented:
Turning of the both untill I'm back from holidays 13 Jan. And turning this PR to draft.

laraPPr commented on test_suite.sh:
```diff
@@ -73,18 +73,22 @@ fi
 echo "EESSI_SOFTWARE_SUBDIR_OVERRIDE: $EESSI_SOFTWARE_SUBDIR_OVERRIDE"
 
 source $TOPDIR/init/bash
+export MODULEPATH=/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/accel/nvidia/cc80/modules/all:$MODULEPATH
```
The environment is set up on the host of the bot and not in the job environment so it is not finding the GPU drivers in the job environment.

laraPPr commented on test_suite.sh:
```diff
@@ -73,18 +73,23 @@ fi
 
 TMPDIR=$(mktemp -d)
 
+ehco "HOSTNAME=$HOSTNAME"
```
```suggestion
echo "HOSTNAME=$HOSTNAME"
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: help

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80
# The bot is now configured to request a full node. This might take a while.

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
Job 15441242 is running now with this config:
```
# reframe_config_bot.py

from eessi.testsuite.common_config import common_logging_config
from eessi.testsuite.constants import *  # noqa: F403

site_configuration = {
    'systems': [
        {
            'name': 'BotBuildTests',  # The system HAS to have this name, do NOT change it
            'descr': 'Software-layer bot',
            'hostnames': ['.*'],
            'modules_system': 'lmod',
            'partitions': [
                {
                    'name': 'x86_64_amd_zen3_nvidia_cc80',
                    'scheduler': 'local',
                    'launcher': 'mpirun',
                    'access': ['--export=None', '--nodes=1', '--cluster=accelgor',]
                    'environs': ['default'],
                    'features': [
                        FEATURES[GPU]
                    ] + list(SCALES.keys()),
                    'resources': [
                        {
                            'name': '_rfm_gpu',
                            'options': ['--gpus-per-node={num_gpus_per_node}'],
                        },
                        {
                            'name': 'memory',
                            'options': ['--mem={size}'],
                        }
                    ],
                    'extras': {
                        # Make sure to round down, otherwise a job might ask for more mem than is available
                        # per node
                        'mem_per_node': __MEM_PER_NODE__,
                        GPU_VENDOR: GPU_VENDORS[NVIDIA],
                    },
                    'devices': [
                        {
                            'type': DEVICE_TYPES[GPU],
                            # Since we specified --gpus-per-node 1, we pretend this virtual partition only has 1 GPU
                            # per node
                            'num_devices': 1,
                        }
                    ],
                    'max_jobs': 1
                    }
                ]
            }
        ],
    'environments': [
        {
            'name': 'default',
            'cc': 'cc',
            'cxx': '',
            'ftn': ''
            }
        ],
    'general': [
        {
            'purge_environment': True,
            'resolve_module_conflicts': False,  # avoid loading the module before submitting the job
            'remote_detect': True,
        }
    ],
    'logging': common_logging_config(),
}
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
It found the config but their was a missing comma. Added it so now retriggering.

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
Looked over a placeholder value in the config

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` and accelerator `nvidia/cc80` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2025.01/pr_842/15441429`
> 
> date	job status	comment
> Jan 20 07:39:45 UTC 2025	submitted	job id `15441429` awaits release by job manager
> Jan 20 07:41:00 UTC 2025	released	job awaits launch by Slurm scheduler
> Jan 20 08:31:11 UTC 2025	running	job `15441429` is running
> Jan 20 08:41:38 UTC 2025	finished	
> 😢 FAILURE _(click triangle for details)_
> Jan 20 08:41:38 UTC 2025	test result	
> 😢 FAILURE _(click triangle for details)_

The disk space was exeeded

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented on test_suite.sh:
```diff
@@ -132,56 +133,73 @@ else
 
+# Get the correct partition name
+REFRAME_PARTITION_NAME=${EESSI_SOFTWARE_SUBDIR//\//_}
+if [ ! -z "$EESSI_ACCELERATOR_TARGET" ]; then
+    REFRAME_PARTITION_NAME=${REFRAME_PARTITION_NAME}_${EESSI_ACCELERATOR_TARGET//\//_}
+fi
```
This is failing for some reason at Gent

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented on test_suite.sh:
```diff
@@ -190,9 +227,9 @@ export PSM3_DEVICES='self,shm'  # this is enough, since we only run single node
 # Check we can run reframe
 reframe --version
 if [[ $? -eq 0 ]]; then
-    echo_green "Succesfully ran 'reframe --version'"
```
Why are the indents here removed as well?

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented on test_suite.sh:
```diff
@@ -217,7 +254,18 @@ else
+if [ -z "$REFRAME_ADDITIONAL_ARGS" ]; then
+    REFRAME_ADDITIONAL_ARGS=""
+fi
+export REFRAME_ARGS="${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} ${REFRAME_ADDITIONAL_ARGS} --nocolor"
```
```suggestion
export REFRAME_ARGS="${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} --nocolor ${REFRAME_NAME_ARGS}"
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented on test_suite.sh:
```diff
@@ -217,7 +254,18 @@ else
+if [ -z "$REFRAME_ADDITIONAL_ARGS" ]; then
+    REFRAME_ADDITIONAL_ARGS=""
+fi
+export REFRAME_ARGS="${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} --nocolor ${REFRAME_NAME_ARGS}"
```
```suggestion
export REFRAME_ARGS="${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} ${REFRAME_ADDITIONAL_ARGS} --nocolor ${REFRAME_NAME_ARGS}"
```

laraPPr commented on test_suite.sh:
```diff
@@ -217,7 +254,18 @@ else
+if [ -z "$REFRAME_ADDITIONAL_ARGS" ]; then
+    REFRAME_ADDITIONAL_ARGS=""
+fi
+export REFRAME_ARGS="${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} ${REFRAME_ADDITIONAL_ARGS} --nocolor ${REFRAME_NAME_ARGS}"
```
```suggestion
export REFRAME_ARGS="--nocolor ${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} ${REFRAME_ADDITIONAL_ARGS} ${REFRAME_NAME_ARGS}"
```

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` and accelerator `nvidia/cc80` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2025.01/pr_842/15441772`
> 
> date	job status	comment
> Jan 22 11:14:06 UTC 2025	submitted	job id `15441772` awaits release by job manager
> Jan 22 11:15:26 UTC 2025	released	job awaits launch by Slurm scheduler
> Jan 22 11:47:37 UTC 2025	running	job `15441772` is running
> Jan 22 12:13:07 UTC 2025	finished	
> 😁 SUCCESS _(click triangle for details)_
> Jan 22 12:13:07 UTC 2025	test result	
> 😢 FAILURE _(click triangle for details)_
> _Reason_
> Failed for unknown reason
> _ReFrame Summary_
> [ PASSED ] Ran 0/0 test case(s) from 0 check(s) (0 failure(s), 0 skipped, 0 aborted)
> _Details_
> ✅ job output file `slurm-15441772.out`
> ✅ no message matching `ERROR: `
> ✅ no message matching `[\s*FAILED\s*].*Ran .* test case`

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
> New job on instance `eessi-bot-vsc-ugent` for CPU micro-architecture `x86_64-amd-zen3` and accelerator `nvidia/cc80` for repository `eessi.io-2023.06-software` in job dir `/scratch/gent/vo/002/gvo00211/SHARED/jobs/2025.01/pr_842/15441780`
> 
> date	job status	comment
> Jan 22 13:29:21 UTC 2025	submitted	job id `15441780` awaits release by job manager
> Jan 22 13:31:21 UTC 2025	released	job awaits launch by Slurm scheduler
> Jan 22 16:53:53 UTC 2025	running	job `15441780` is running
> Jan 22 17:17:23 UTC 2025	finished	
> 😁 SUCCESS _(click triangle for details)_
> Jan 22 17:17:23 UTC 2025	test result	
> 😢 FAILURE _(click triangle for details)_

The problem is now that the arguments are in brackets `'--tag=CI --tag=1_4_node'` which for some reason breaks the filtering but this is not the case for the `--name` filtering. So I will have a deeper look how to get around this but that will be for next week.

laraPPr commented on bot/test.sh:
```diff
@@ -50,11 +50,6 @@ cat ${JOB_CFG_FILE}
 echo "bot/test.sh: obtaining configuration settings from '${JOB_CFG_FILE}'"
 cfg_load ${JOB_CFG_FILE}
 
-# if http_proxy is defined in ${JOB_CFG_FILE} use it, if not use env var $http_proxy
-HTTP_PROXY=$(cfg_get_value "site_config" "http_proxy")
-HTTP_PROXY=${HTTP_PROXY:-${http_proxy}}
-echo "bot/test.sh: HTTP_PROXY='${HTTP_PROXY}'"
-
```
I'm not sure why this is double but I think one of them can go? @casparvl @trz42

casparvl replied:
Must be a mistake yes. Maybe introduced by me when we created `test.sh` from selective stuff in `build.sh`

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
Ok this fixed it not being able to handle `--tag` but now it can't hand `-n EESSI_OSU -n EESSI_LAMMPS` so trying something else let's see

bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
@trz42 I have a question I just figured out that the token of the bot is expired but it keeps working is this normal? No right?

boegel commented:
> @trz42 I have a question I just figured out that the token of the bot is expired but it keeps working is this normal? No right?

A GitHub token is currently required by the bot to start, but isn't actually used, it uses a mechanism to request temporary short-lived tokens instead. So the corresponding variable must be defined, but it canbe anything (an actual valid or expired token, or even just garbabe...)

casparvl commented on test_suite.sh:
```diff
@@ -85,6 +85,26 @@ source $TOPDIR/init/bash
 # Reason is that the LMOD cache is normally only updated on the Stratum 0, once everything is ingested
 export LMOD_IGNORE_CACHE=1
 
+software_layer_dir=$(dirname $(realpath $0))
```
This whole section up to `echo "test_suite.sh: EESSI_ACCELERATOR_TARGET='${EESSI_ACCELERATOR_TARGET}'"` can now be removed again, now that the EESSI_ACCELERATOR_TARGET is set in `test.sh`, right?

casparvl commented on test_suite.sh:
```diff
@@ -217,11 +254,25 @@ else
+echo ${REFRAME_ARGS}
+echo "${REFRAME_ARGS}"
+echo $REFRAME_ARGS
+reframe ${REFRAME_CI_TAG} ${REFRAME_SCALE_TAG} ${REFRAME_ADDITIONAL_ARGS} --nocolor "${REFRAME_NAME_ARGS}" --list -v
```
wait, so... there was no way to get `reframe ${REFRAME_ARGS}` to work (with _any_ choice of quotes/braces)? Keeping that in a single environment variable was done to ensure that `--list` and `--run` would be called with the exact same arguments... The current way of doing things opens up the possibility of someone adding something to the argument list when calling `--list` but forgetting to do so for the `--run`.

laraPPr commented on bot/test.sh:
```diff
@@ -123,6 +118,12 @@ mkdir -p ${STORAGE}
 
+echo "bot/test.sh: MODULEPATH='${MODULEPATH}'"
+# Checking if it is running at Gent because than the modules need to be unloaded
+if [ '$HPCUGENT_FAMILY_CLUSTER' ]; then
```
Maybe I can up a `module --force purge` in the .bashrc @boegel any objections?

boegel replied:
That's a bit dirty, but better than leaking system-specific stuff in here I think...

So go ahead, but add a clear comment to explain why that's done.
And we should track the `.bashrc` we're using in Git somewhere, because it's no doubt going to evolve...

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: help

laraPPr commented:
bot: show_config

laraPPr commented:
bot: build instance:eessi-bot-vsc-ugent repo:eessi.io-2023.06-software accel:nvidia/cc80

laraPPr commented:
bot: help

laraPPr commented:
Was testing the build and deploy on a new cluster. This pr can now be closed without merging.
